### DISCUSSION TITLE
Help: Specific layout with Grid/Flex that does not pop out under its max height

### DISCUSSION THREAD

**Discussion by bennettdams** [tailwindlabs/tailwindcss#2480] [Help] [answered]
Hey!

I'm trying to do a specific layout together with React components.

Here is a CodeSandbox to try it out:   https://codesandbox.io/s/eloquent-hopper-0nczm?file=/src/App.tsx

Imagine this page, done via grid (no scroll):

![image](https://user-images.githubusercontent.com/29319414/94916683-7265e380-04af-11eb-84bf-9fed2fcce82e.png)

The green box is a legend which is filled with categories. Inside the categories there are buttons (blue boxes):
This shows how I want it to be:

![image](https://user-images.githubusercontent.com/29319414/94911450-73464780-04a6-11eb-8745-f959f09d9a46.png)

Take category 2 as an example: As soon as the buttons fill the height of the green box, they should take the next "column" (where box no. 5 starts now).

In reality, they just fill up ONE column and therefore ignore the max height and just pop out under.

How to I make them wrap/take the next column of their available space?

These are components btw, so I don't want to hard-code the behavior of the categories/buttons.

Any help is appreciated!

**Comment by daniel-norris** [comment]
Why not set the overall 3 part layout of your app with grid. Then use flexbox to cover the the button functionality at the bottom. 

You can then use flexbox wrap to achieve what you want (I think).

https://css-tricks.com/almanac/properties/f/flex-wrap/

**Reply by bennettdams** [reply]
Thanks for your input. If you have a look at the CodeSandbox you can see that I already use `flex-wrap` for that part, but exactly that is the problem: The boxes are not wrapping but rather taking more vertical space than they should.

**Reply by MartinHughes-BPC** [reply]
@bennettdams Hey Bennet,

I think I got this working:

In this bit remove the flex-1
```
const LegendButton = ({ no }: { no: number }) => {
  return (
    <div className="**flex-1** border-2 w-16 border-blue-300 text-xs">
      Button {no}
    </div>
  );
};
```
And here:

```
return (
    <div className="flex-1 p-4 border-4 border-orange-300 h-full">
      <p>{title}</p>
      <div className="flex flex-col flex-wrap **h-full**">{children}</div>
    </div>
  );
```
Add h-full

... and that should do the trick.

Cheers,
Martin.

**Reply by bennettdams** [reply]
That's perfect, thanks @MartinHughes-BPC! Could you post this as a new comment, so I can mark it as an answer?

**Reply by MartinHughes-BPC** [reply]
Will do!

**Comment by MartinHughes-BPC** [comment, marked as answer]
I think I got this working:

In this bit remove the flex-1
```
const LegendButton = ({ no }: { no: number }) => {
  return (
    <div className="**flex-1** border-2 w-16 border-blue-300 text-xs">
      Button {no}
    </div>
  );
};
```
And here:

```
return (
    <div className="flex-1 p-4 border-4 border-orange-300 h-full">
      <p>{title}</p>
      <div className="flex flex-col flex-wrap **h-full**">{children}</div>
    </div>
  );
```
Add h-full

... and that should do the trick.

Cheers,
Martin.
